### PR TITLE
Add GRetriever RelBench example demo and from_relbench integration

### PR DESCRIPTION
**Pull request by AJamal27891**
Adds `examples/llm/relbench_gretriever.py`, demo the integration of `from_relbench` utility output with `GRetriever`.

The example highlights a key pattern for heterogeneous-to-homogeneous conversion: calling `to_homogeneous()` on raw RelBench-derived `HeteroData` can lose node features or force padding when some node types have no attributes. To avoid this, all node types are first projected into a shared embedding space via a `HeteroFeatureProjector` (using `HeteroDictLinear` for feature-bearing types and `nn.Embedding` for featureless ones), and only then converted.

This pattern preserves node features and enables reliable use of heterogeneous graphs in homogeneous GNN pipelines such as `GRetriever`.

Includes:

* End-to-end example from RelBench database → `HeteroData` → projected features → homogeneous graph → `GRetriever`
* Handling of featureless node types via learned embeddings
* Minimal training and inference loop for demo

Depends on [#10628](https://github.com/pyg-team/pytorch_geometric/pull/10628).  
Ref: [#10353](https://github.com/pyg-team/pytorch_geometric/pull/10353).  
Partially Closes issue [#9839](https://github.com/pyg-team/pytorch_geometric/issues/9839).

### CODE REVIEW

**Review by puririshi98 — DISMISSED** [review]
@AJamal27891 can you include a log of an e2e run of the example added?

**Comment by AJamal27891** [comment]
> @AJamal27891 can you include a log of an e2e run of the example added?

```bash 
root@64af3f32cfc8:/workspace# python /workspace/pytorch_geometric/examples/llm/relbench_gretriever.py --epochs 10 --llm Qwen/Qwen2-0.5B
Loading RelBench rel-f1 dataset...
Loading Database object from /root/.cache/relbench/rel-f1/db...
Done in 0.07 seconds.
Graph: 9 node types, 26 edge types
Homogeneous: edge_index=[2, 338842]

Initializing GRetriever with LLM=Qwen/Qwen2-0.5B...
Setting up 'Qwen/Qwen2-0.5B' with configuration: {'revision': 'main', 'max_memory': {0: '3GiB'}, 'low_cpu_mem_usage': True, 'device_map': 'auto', 'dtype': torch.bfloat16}
/usr/local/lib/python3.12/dist-packages/torch/library.py:357: UserWarning: Warning only once for all operators,  other operators may also be overridden.
  Overriding a previously registered kernel for the same operator and the same dispatch key
  operator: flash_attn::_flash_attn_backward(Tensor dout, Tensor q, Tensor k, Tensor v, Tensor out, Tensor softmax_lse, Tensor(a6!)? dq, Tensor(a7!)? dk, Tensor(a8!)? dv, float dropout_p, float softmax_scale, bool causal, SymInt window_size_left, SymInt window_size_right, float softcap, Tensor? alibi_slopes, bool deterministic, Tensor? rng_state=None) -> Tensor
    registered at /usr/local/lib/python3.12/dist-packages/torch/_library/custom_ops.py:926
  dispatch key: ADInplaceOrView
  previous kernel: no debug info
       new kernel: registered at /usr/local/lib/python3.12/dist-packages/torch/_library/custom_ops.py:926 (Triggered internally at /opt/pytorch/pytorch/aten/src/ATen/core/dispatch/OperatorEntry.cpp:208.)
  self.m.impl(
Skipping import of cpp extensions due to incompatible torch version 2.10.0a0+a36e1d39eb.nv26.01.42222806 for torchao version 0.15.0             Please see https://github.com/pytorch/ao/issues/2919 for more info
Using device: cuda:0

Training 10 epochs on 4 samples...
Epoch 01: Loss=3.5860
Epoch 02: Loss=2.2119
Epoch 03: Loss=0.9973
Epoch 04: Loss=1.1809
Epoch 05: Loss=0.6012
Epoch 06: Loss=1.3658
Epoch 07: Loss=0.2758
Epoch 08: Loss=0.2540
Epoch 09: Loss=0.4419
Epoch 10: Loss=0.5418

Inference:
Q: Which entity types appear in this Formula 1 graph?
A: The graph contains node types such as drivers, constructors circuits races, qualifying edges.

Q: Why do we project all node types before calling to_homogeneous?
A: The projection creates a shared embedding space so GRetriever can process the graph as a single homogeneous tensor.
```

**Review by akihironitta — COMMENTED** [review]
@AJamal27891 Sorry about the delay. ~Could we make this PR change on top of #10628?~ Looks like GitHub doesn't allow contributors to stack PRs submitted to upstream repo. I went ahead and merged https://github.com/pyg-team/pytorch_geometric/pull/10628 already, so there's no action required.

**Review by akihironitta — APPROVED** [review]
@AJamal27891 Great work! I pushed a commit to fix some issue, but LGTM!

**Review by puririshi98 — APPROVED** [review]
